pkg/fileutil/files.go

```diff
@@ -244,58 +257,77 @@ func CreateFile(p string) (*os.File, error) {
 	return os.Create(p)
 }
 
-func UploadFile(filePath, targetURL, formFileFieldName string, headers map[string]string) error {
-	file, err := os.Open(filePath)
+func UploadFile(args *UploadFileArguments) error {
+	file, err := os.Open(args.FilePath)
 	if err != nil {
 		return fmt.Errorf("error opening file: %w", err)
 	}
 	defer file.Close()
 
-	// Create a new buffer to store the multipart/form-data request
-	body := &bytes.Buffer{}
-	writer := multipart.NewWriter(body)
+	retryDelayInMS := args.InitialDelayInMS
+	currentUploadError := err
 
-	// Create a form field for the file
-	fileWriter, err := writer.CreateFormFile(formFileFieldName, filePath)
-	if err != nil {
-		return fmt.Errorf("error creating form file: %w", err)
-	}
-
-	// Copy the file content into the form field
-	_, err = ioCopy(fileWriter, file)
-	if err != nil {
-		return fmt.Errorf("error copying file content: %w", err)
-	}
+	for i := 1; i <= args.MaxTries; i++ {
+		// exponential backoff
+		time.Sleep(time.Duration(retryDelayInMS) * time.Millisecond)
+		retryDelayInMS *= args.BackoffFactor
+		fmt.Print(args.RetryDisplayMessage + "\n")
 
-	// Close the multipart writer to finalize the request
-	writer.Close()
-
-	req, err := newRequestWithContext(http_context.Background(), "POST", targetURL, body)
-	if err != nil {
-		return err
-	}
+		// Create a new buffer to store the multipart/form-data request
+		body := &bytes.Buffer{}
+		writer := multipart.NewWriter(body)
+		// Create a form field for the file
+		fileWriter, err := writer.CreateFormFile(args.FormFileFieldName, args.FilePath)
+		if err != nil {
+			return fmt.Errorf("error creating form file: %w", err)
+		}
+		// Copy the file content into the form field
+		_, err = ioCopy(fileWriter, file)
+		if err != nil {
+			return fmt.Errorf("error copying file content: %w", err)
+		}
+		// Close the multipart writer to finalize the request
+		writer.Close()
 
-	headers["Content-Type"] = writer.FormDataContentType()
+		headers := args.Headers
+		headers["Content-Type"] = writer.FormDataContentType()
+		req, err := newRequestWithContext(http_context.Background(), "POST", args.TargetURL, body)
+		if err != nil {
+			currentUploadError = err
+			continue
+		}
 
-	// set headers
-	for k, v := range headers {
-		req.Header.Set(k, v)
-	}
+		// set headers
+		for k, v := range headers {
+			req.Header.Set(k, v)
+		}
+		client := &http.Client{}
+		response, err := client.Do(req)
+		if err != nil {
+			currentUploadError = err
+			// If we have a dial tcp error, we should retry with exponential backoff
+			continue
+		}
+		data, _ := io.ReadAll(response.Body)
+		responseStatusCode := response.StatusCode
+		response.Body.Close()
+
+		// Return success for 2xx status code
+		if response.StatusCode == http.StatusOK {
+			currentUploadError = nil
+			fmt.Print("upload successful\n")
```

Comment:
```suggestion
			fmt.Println("upload successful")
```

Reply:
Yeah, I noticed it later. I fixed it in the other PR when I was working on it. https://github.com/astronomer/astro-cli/pull/1518 This one
@kushalmalani